src/ngraph/runtime/backend.hpp

```diff
@@ -79,8 +80,8 @@ class ngraph::runtime::Backend
 
     /// \brief Compiles a Function.
     /// \param func The function to compile
-    /// \returns true if compile is successful, false otherwise
-    virtual bool compile(std::shared_ptr<Function> func) = 0;
+    /// \returns compiled function or nullptr on failure
+    virtual Handle compile(std::shared_ptr<Function> func) = 0;
```

Comment:
How is the compilation error conveyed to the caller? Exceptions I suppose. Please provide some information about it in the comment for this function.

Reply:
There are two things I would like to suggest.
1. We keep the `call(...)` API as is and internally that function calls `compile()` (instead of the error thrown). 
2. We change the signature to `call()` so that it now takes a `Handle` as opposed to a function. That way - caller will call compile and then call. Can we rename `call()` to `execute()`? We are really executing the function - no?
I prefer the second option as it enforced the caller to call `compile()` first.

Reply:
Your second option is what I did in the last giant PR that you wanted split into smaller PRs. This is part 1 of several PRs that will ultimately change call to be `call(Handle)`
The point of this PR is to enforce that the unit tests all call compile prior to calling call.
I agree that execute() is a much better name than call() but that should be a separate PR.

Reply:
Sounds great!